src/net.c

```diff
@@ -554,7 +554,7 @@ coap_send_ack(coap_context_t *context,
   return result;
 }
 
-#if defined(WITH_POSIX) || defined(WITH_CONTIKI)
+#if !defined(WITH_LWIP)
```

Comment:
Please don't use negative logic here and let that check as it is, it's hard to figure out some not well knowledge people what !WITH_LWIP means!